include/fastdds/rtps/messages/RTPSMessageGroup.h

```diff
@@ -252,6 +252,17 @@ class RTPSMessageGroup
         return current_sent_bytes_ + full_msg_->length;
     }
 
+    Endpoint* get_current_endpoint()
+    {
+        return endpoint_;
+    }
+
+    void reset_current_endpoint()
+    {
+        endpoint_ = nullptr;
+        flush_and_reset();
```

Comment:
Hi. I used `flush_and_reset`, because in general, the behavior is different, at least due to [this string](https://github.com/shumov-ag/Fast-DDS/blob/fix-deadlock-between-flowcontroller-and-matched_reader_remove/src/cpp/rtps/messages/RTPSMessageGroup.cpp#L318).

Reply:
> do you think you could add a test for this specific deadlock?

I think I can try to do that

Reply:
> Is this intentional?

Yes. Since endpoint_ is nullptr, we only need to do complete reset.